### PR TITLE
added chain_api: chain_get_header, chaing_get_block, chain_get_finalized_head

### LINKED ISSUE #50: get block api
'''rust
pub fn chain_get_block(hash:String) -> Value {
    json!({
    "method": "chain_getBlock",
    "params": [hash],
    "jsonrpc": "2.0",
    "id":"1"
    })
}
'''
Any support for rpc chain_getBlock?

### PR DESCRIPTION
**Pull request by clangenb**
* Extended rpc api
* closes #50

### CODE REVIEW

**Comment by clangenb** [comment]
ci fails as it tries to run the example against a more recent node

File: src/lib.rs [hunk 1]
```diff
@@ -204,6 +203,18 @@ impl<P> Api<P>
         hexstr_to_u256(result_str).unwrap()
     }
 
+    pub fn get_finalized_head(&self) -> WsResult<String> {
```
Comment (brenzi) [inline src/lib.rs]: String result isn't very nice. can we decode this to `Header`? I guess we could use the generic header in `use sr_primitives::generic::Header;`

Comment (brenzi) [reply src/lib.rs]:
```
let ret = api.get_header(None).unwrap();
let hdr: Header = serde_json::from_str(&ret).unwrap();
```

**Comment by brenzi** [comment]
included in PR #66

**Pull request closed without merging**